Plutarch/DataRepr/Internal/Field.hs

```diff
@@ -186,6 +226,21 @@ type family TermsOf (s :: S) (as :: [PLabeledType]) :: [Type] where
   TermsOf _ '[] = '[]
   TermsOf s ((name ':= a) ': as) = (Labeled name (Term s (PAsData a))) ': TermsOf s as
 
+-- | Get the index of (either kind of) labeled term
+type LabelIndex :: Symbol -> [k] -> Nat
+type family LabelIndex (name :: Symbol) (as :: [k]) :: Nat where
+  LabelIndex name ((name ':= _) ': _) = 0
+  LabelIndex name ((Labeled name _) ': _) = 0
+  LabelIndex name (_ ': as) = (LabelIndex name as) + 1
+
+type AllIndices :: [Symbol] -> [k] -> [Nat]
+type family AllIndices (ns :: [Symbol]) (as :: [k]) :: [Nat] where
+  AllIndices '[] _ = '[]
+  AllIndices (n ': ns) as = (LabelIndex n as) ': (AllIndices ns as)
+
+type family FieldsRange (ns :: [Symbol]) (as :: [k]) :: (Nat, Nat) where
+  FieldsRange ns as = FindMinMax (AllIndices ns as)
```

Comment:
There's now *a lot* of type-family evaluation going on... with the nesting, and list-recursion the type errors are pretty crazy